### PR TITLE
Changes for Cloud apps support

### PR DESCRIPTION
**Pull request by sathaley1**
Changes to Commcell request method to support XML for QCommand operations.

Changes in Subclient.py: Agent based subclient content getter and setter.

### CODE REVIEW

File: cvpysdk/subclient.py [hunk 1]
```diff
@@ -169,6 +169,40 @@ def _get_subclients(self):
...
+        content = []
+
+        if self._backupset_object._agent_object.agent_name == "file system":
+            for path in subclient_content:
+                file_system_dict = {
+                    "path": path
+                }
+
+                content.append(file_system_dict)
+        elif self._backupset_object._agent_object.agent_name == "cloud apps":
```
Comment (klogan12) [inline cvpysdk/subclient.py]: Hi Siddharth,

For each Agent , we thought of creating one class per sub-client that will implement the abstract class. 
You can check with Jatin. I will inbox you the design.

Comment (sathaley1) [inline cvpysdk/subclient.py]: Sure Karthik, Please send me the design so I can make changes accordingly. Meanwhile I will get in touch with Jatin.Thanks!

File: cvpysdk/subclient.py [hunk 2]
```diff
@@ -206,11 +240,14 @@ def add(self, subclient_name, content, storage_policy, description=''):
                     if response is empty
                     if response is not success
         """
-        if not (isinstance(subclient_name, str) or
-                isinstance(content, list) or
+        if not (isinstance(subclient_name, str) and
+                isinstance(content, list) and
```
Comment (GoelJatin) [inline cvpysdk/subclient.py]: Hi Siddharth,

Thanks for pointing out the bug here.

**Comment by sathaley1** [comment]
Closing this request as there have been design changes requested by the hosts. I will open another pull request with required changes.

**Pull request closed without merging**